pkg/provisioner/ironic/ironic.go

```diff
@@ -1623,3 +1636,58 @@ func (p *ironicProvisioner) IsReady() (result bool, err error) {
 	checker := newIronicDependenciesChecker(p.client, p.inspector, p.log)
 	return checker.IsReady()
 }
+
+func (p *ironicProvisioner) IsProvisioning() bool {
+	_, ok := p.getCurrentProvisioningHosts()[p.host.Name]
+	return ok
+}
+
+func (p *ironicProvisioner) CanBeProvisioned() bool {
+	return len(p.getCurrentProvisioningHosts()) < maxProvisioningHosts
+}
+
+func (p *ironicProvisioner) getCurrentProvisioningHosts() map[string]struct{} {
+	//lazy init
+	if p.provisioningHosts != nil {
+		return p.provisioningHosts
+	}
+
+	p.provisioningHosts = make(map[string]struct{})
+	if err := p.loadProvisioningHosts(p.provisioningHosts); err != nil {
+		p.log.Error(err, "Unable to get hosts currently being provisioned")
+	}
+
+	return p.provisioningHosts
+}
+
+func (p *ironicProvisioner) loadProvisioningHosts(hosts map[string]struct{}) error {
+
+	pager := nodes.List(p.client, nodes.ListOpts{
+		Fields: []string{"uuid,name,provision_state,driver_internal_info"},
+	})
+	if pager.Err != nil {
+		return pager.Err
+	}
+
+	page, err := pager.AllPages()
+	if err != nil {
+		return err
+	}
+
+	allNodes, err := nodes.ExtractNodes(page)
+	if err != nil {
+		return err
+	}
+
+	for _, node := range allNodes {
+
+		switch node.ProvisionState {
+		case "cleaning", "clean wait", "inspecting", "inspect wait",
+			"deploying", "wait call-back":
```

Comment:
Use the constants defined in the nodes package.

Reply:
Initially I couldn't use the const since some of the states were missing (I've also added them in this PR: https://github.com/gophercloud/gophercloud/pull/2052). But then I moved to a more restricted subset of states, so if we keep such subset (the original proposed was slightly larger) it would be ok to use the const. Done